.pre-commit-config.yaml

```diff
@@ -0,0 +1,49 @@
+# This file contains 2 stages
+# pre-commit stage - runs by default pre-commit
+# manual stage - runs the same tools but modifies files
+# run as: pre-commit run --all-files --hook-stage manual
```

Comment:
You should add a note here that the contents of this file should be kept in sync with the `Run python style and linting tools` step in `.github/workflows/python_lint_and_test.yaml`.  (And, it might be worth adding a note there pointing to the configuration here....)